src/classes/SRVC_Contact_PrimaryAffiliations.cls

```diff
@@ -87,27 +88,30 @@ public virtual with sharing class SRVC_Contact_PrimaryAffiliations {
     @TestVisible
     private List<Affiliation__c> getAffiliationsToCreateForContact(Contact contact, List<MODL_PrimaryAfflMapper> afflMappingsModelsList) {
         List<Affiliation__c> affiliationsToCreateForEachContactList = new List<Affiliation__c>();
+
         try {
             for (MODL_PrimaryAfflMapper eachModel: afflMappingsModelsList) {
                 if (contact.get(eachModel.getContactFieldApiName()) == null) {
                     continue;
                 }
 
                 if (eachModel.getIsValid() == false){
-                    contact.addError(Label.afflAccoutMappingError);
-                    throw new ERR_ExceptionHandler.AffAccountRecordTypeMappingsException(Label.afflAccoutMappingError);
+                    contact.addError(eachModel.getInvalidErrorMessage());
+                    throw new ERR_ExceptionHandler.AffAccountRecordTypeMappingsException(eachModel.getInvalidErrorMessage());
                 }
 
                 Affiliation__c afflToCreate;
                 
                 try {
                     afflToCreate = new Affiliation__c(Contact__c = contact.ID, Account__c = (Id)contact.get(eachModel.getContactFieldApiName()), Primary__c = true);
                 } catch (System.TypeException typeExc) {
-                    Contact.addError(typeExc.getMessage());
-                    throw new ERR_ExceptionHandler.AffAccountRecordTypeMappingsException(Label.afflAccoutMappingError);
+                    // mapped contact field is valid, but value provided is not a Lookup to a contact Id
+                    contact.addError(eachModel.getInvalidErrorMessage());
+                    throw new ERR_ExceptionHandler.AffAccountRecordTypeMappingsException(eachModel.getInvalidErrorMessage());
                 } catch (System.StringException stringExc) {
-                    Contact.addError(stringExc.getMessage());
-                    throw new ERR_ExceptionHandler.AffAccountRecordTypeMappingsException(Label.afflAccoutMappingError);
+                    // mapped contact field is valid, but value provided is not a string 
+                    contact.addError(eachModel.getInvalidErrorMessage());
+                    throw new ERR_ExceptionHandler.AffAccountRecordTypeMappingsException(eachModel.getInvalidErrorMessage());
```

Comment:
Have a quick question on this. Looks like we will never throw this exception right? We are adding error to the contact trigger?

Reply:
I think this exception could occur when the value in getContactFieldApiName is not a valid ID?

Reply:
That's accurate. Depending on the type of field passed in there can be a number of different explicit exceptions we are catching for.